### ISSUE TITLE
[Issue]: LiveTv.Listings.XmlTvListingsProvider duplicates downloaded xml for every single channel while Guide Refresh

### ISSUE THREAD

**Issue by OEvgeny** [jellyfin/jellyfin#7164]
### Please describe your bug

I noticed Guide Refresh takes some time when adding a relatively big XMLTV source for the Guide.

Mine was ~50M gzipped (300 M when not).

I see a lot of duplicate files in `/cache/temp/` in the middle of the Guide refresh, like this:

```sh
# du -h /cache/temp
....
300M    /cache/temp/03fc63d0-aa91-4714-81c0-0c68517efe21
300M    /cache/temp/6c52ffd5-edda-472b-914f-6cd1cdf5c285
300M    /cache/temp/1e566c30-36f2-4722-a790-bb644966d3de
300M    /cache/temp/19599cbc-ae52-4c8c-8b3f-5dae26c8f8e6
300M    /cache/temp/61ef5a23-1269-4da2-b547-f48b9ed03186
300M    /cache/temp/93616e65-a462-49f2-961a-bf61caa7fd18
300M    /cache/temp/ea28bae4-9fe8-44db-9ec7-bb9b6a08bd97
300M    /cache/temp/b1a2eb8d-72e6-4fe8-a2b5-583560a000f3
300M    /cache/temp/21adb16a-0217-40fc-8bc3-84cd2c3624cb
300M    /cache/temp/4e1a9e81-47f6-4f20-888a-0910443658b8
300M    /cache/temp/761c7d17-d0e0-4080-8401-0617daa15d34
300M    /cache/temp/12e1a744-0af2-4e99-90fb-099f0a8cbaac
300M    /cache/temp/1d7e762d-989d-435f-82c2-663535e282a2
300M    /cache/temp/f54d5fb7-1dc0-4bdf-b205-c2b9260dcb7a
300M    /cache/temp/4474a353-7de5-4585-a9dc-e4fdc3341548
300M    /cache/temp/0a58c1e1-e4d5-4257-b87a-c77cdbd398a0
300M    /cache/temp/86db7d23-dced-43ab-89d4-11f5457d88c0
300M    /cache/temp/c46bf0e3-7fba-440a-82b9-38a8a9d53c60
300M    /cache/temp/24f2a5c6-5202-44ae-aa28-91aa3ee9f34c
300M    /cache/temp/3dc04941-307f-4b2a-940f-6e706209b5d7
300M    /cache/temp/38e27e39-6aab-44de-b1b9-051b5fbe1b79
53G     /cache/temp/
```

My PVR has more then 1000 channels, so this approach doesn't scale well to me

Here is the link to the XMLTV epg: http://epg.it999.ru/edem.xml.gz

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Linux
- Virtualization: Docker
- Clients: N/a
- Browser: N/a 
- FFmpeg Version: N/a
- Playback Method: N/a
- Hardware Acceleration: N/a
- Plugins: None
- Reverse Proxy: Caddy
- Base URL: N/a
- Networking: Host
- Storage: local
```

### Jellyfin logs

```shell
storage-jellyfin-1  | [02:58:14] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Getting programs for channel null-Небеса ТВ7 HD from XmlTV-
storage-jellyfin-1  | [02:58:14] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Getting xmltv programs for channel 6030
storage-jellyfin-1  | [02:58:14] [INF] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: xmltv path: http://localhost/files/edem.xml.gz
storage-jellyfin-1  | [02:58:20] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Opening XmlTvReader for /cache/temp/61ef5a23-1269-4da2-b547-f48b9ed03186/edem.xml
storage-jellyfin-1  | [02:58:45] [DBG] [19] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 14.0826ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where type=@type AND ChannelId=@ChannelId
storage-jellyfin-1  | [02:58:45] [DBG] [19] Emby.Server.Implementations.LiveTv.LiveTvManager: Channel Небеса ТВ7 HD has 52 new programs and 0 updated programs
storage-jellyfin-1  | [02:58:46] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Getting programs for channel null-Министерство идей HD from XmlTV-
storage-jellyfin-1  | [02:58:46] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Getting xmltv programs for channel 6031
storage-jellyfin-1  | [02:58:46] [INF] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: xmltv path: http://localhost/files/edem.xml.gz
storage-jellyfin-1  | [02:58:51] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Opening XmlTvReader for /cache/temp/93616e65-a462-49f2-961a-bf61caa7fd18/edem.xml
storage-jellyfin-1  | [02:59:18] [DBG] [19] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 19.7916ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where type=@type AND ChannelId=@ChannelId
storage-jellyfin-1  | [02:59:18] [DBG] [19] Emby.Server.Implementations.LiveTv.LiveTvManager: Channel Министерство идей HD has 374 new programs and 0 updated programs
storage-jellyfin-1  | [02:59:19] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Getting programs for channel null-ТВ Экстра HD from XmlTV-
storage-jellyfin-1  | [02:59:19] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Getting xmltv programs for channel 6036
storage-jellyfin-1  | [02:59:19] [INF] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: xmltv path: http://localhost/files/edem.xml.gz
storage-jellyfin-1  | [02:59:25] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Opening XmlTvReader for /cache/temp/ea28bae4-9fe8-44db-9ec7-bb9b6a08bd97/edem.xml
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 21.1039ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where type=@type AND ChannelId=@ChannelId
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.LiveTv.LiveTvManager: Channel ТВ Экстра HD has 0 new programs and 0 updated programs
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Getting programs for channel null-РБК 50 from XmlTV-
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: EPG channel not found for tuner channel null-РБК 50 from XmlTV-
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 22.2905ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where type=@type AND ChannelId=@ChannelId
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.LiveTv.LiveTvManager: Channel РБК 50 has 0 new programs and 0 updated programs
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Getting programs for channel null-RBC TV HD from XmlTV-
storage-jellyfin-1  | [02:59:54] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Getting xmltv programs for channel 743
storage-jellyfin-1  | [02:59:54] [INF] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: xmltv path: http://localhost/files/edem.xml.gz
storage-jellyfin-1  | [02:59:59] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Opening XmlTvReader for /cache/temp/b1a2eb8d-72e6-4fe8-a2b5-583560a000f3/edem.xml
storage-jellyfin-1  | [03:00:26] [DBG] [19] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 21.8714ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where type=@type AND ChannelId=@ChannelId
storage-jellyfin-1  | [03:00:26] [DBG] [19] Emby.Server.Implementations.LiveTv.LiveTvManager: Channel RBC TV HD has 321 new programs and 0 updated programs
storage-jellyfin-1  | [03:00:27] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Getting programs for channel null-RBK Ekaterinburg from XmlTV-
storage-jellyfin-1  | [03:00:27] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: EPG channel not found for tuner channel null-RBK Ekaterinburg from XmlTV-
storage-jellyfin-1  | [03:00:27] [DBG] [19] Emby.Server.Implementations.Data.SqliteItemRepository: GetItemList query time (slow): 20.0303ms. Query: select type,data,StartDate,EndDate,ChannelId,IsMovie,IsSeries,EpisodeTitle,IsRepeat,CommunityRating,CustomRating,IndexNumber,IsLocked,PreferredMetadataLanguage,PreferredMetadataCountryCode,Width,Height,DateLastRefreshed,Name,Path,PremiereDate,Overview,ParentIndexNumber,ProductionYear,OfficialRating,ForcedSortName,RunTimeTicks,Size,DateCreated,DateModified,guid,Genres,ParentId,Audio,IsInMixedFolder,DateLastSaved,LockedFields,Studios,Tags,OriginalTitle,PrimaryVersionId,DateLastMediaAdded,Album,CriticRating,IsVirtualItem,SeriesName,PresentationUniqueKey,InheritedParentalRatingValue,ExternalSeriesId,Tagline,ProviderIds,Images,ProductionLocations,ExtraIds,TotalBitrate,ExtraType,ExternalId,SeriesPresentationUniqueKey,ShowId,OwnerId from TypedBaseItems A where type=@type AND ChannelId=@ChannelId
storage-jellyfin-1  | [03:00:27] [DBG] [19] Emby.Server.Implementations.LiveTv.LiveTvManager: Channel RBK Ekaterinburg has 0 new programs and 0 updated programs
storage-jellyfin-1  | [03:00:27] [DBG] [19] Emby.Server.Implementations.LiveTv.EmbyTV.EmbyTV: Getting programs for channel null-Инфоканал from XmlTV-
storage-jellyfin-1  | [03:00:27] [DBG] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Getting xmltv programs for channel 2264
storage-jellyfin-1  | [03:00:27] [INF] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: xmltv path: http://localhost/files/edem.xml.gz
storage-jellyfin-1  | [03:00:27] [INF] [19] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Downloading xmltv listings from http://localhost/files/edem.xml.gz
storage-jellyfin-1  | [03:00:33] [DBG] [23] Emby.Server.Implementations.LiveTv.Listings.XmlTvListingsProvider: Opening XmlTvReader for /cache/temp/21adb16a-0217-40fc-8bc3-84cd2c3624cb/edem.xml
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by OEvgeny** [comment]
Temporary fixed the issue on my end:
- Added a tmpfs volume for `/cache/temp` so it doesn't bug a disk. This is a nice to have thing in a default docker setup as well.
- Setup automatic cleanup for temp files, so it doesn't grow beyond temp fs size limit

**Comment by OEvgeny** [comment]
Closing as the issue  didn't get any attention from community or maintainers

**Comment by OEvgeny** [comment]
Closing as the issue didn't get any attention from community or maintainers

**Comment by Bond-009** [comment]
I think your issue is fixed by one of these PRs: #7567 #7592 #7625
These should be included in the latest 10.8 beta, so you can test for yourself.
If you think the issue hasn't been fixed you can reopen this issue.
